File: ios/Modules/LiveStream/LiveStreamViewManager.m
//
//  LiveStreamViewManager.m
//  liveStremSDKIos
//
//  Created by Nguyễn Anh Tuấn on 20/2/25.
//
#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(LiveStreamViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(roomId, NSString)
RCT_EXTERN_METHOD(leaveRoom:(nonnull NSNumber *)reactTag)
RCT_EXTERN_METHOD(getInfoRoom:(nonnull NSNumber *)reactTag)

@end

File: ios/Modules/LiveStream/LiveStreamViewManager.swift
//
//  LiveStreamViewManager.swift
//  liveStremSDKIos
//
//  Created by Nguyễn Anh Tuấn on 20/2/25.
//

import Foundation
import React

@objc(LiveStreamViewManager)
class LiveStreamViewManager: RCTViewManager {
  var liveStreamView: LiveStreamView!
  override func view() -> UIView! {

  
    liveStreamView = LiveStreamView()
    return liveStreamView
  }

  @objc func leaveRoom(_ reactTag: NSNumber) {
    liveStreamView.leaveRoom()
  }
  

  @objc override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
